### ISSUE TITLE
prefer-destructuring reports false positives with compound assignments

### ISSUE THREAD

**Issue by fasttime** [eslint/eslint#7881]
**Tell us about your environment**

* **ESLint Version:** 3.13.0
* **Node Version:** 7.3.0
* **npm Version:** 4.0.5

**What parser (default, Babel-ESLint, etc.) are you using?**

default

**Please show your full configuration:**

```
{ "parserOptions": { "ecmaVersion": 6 } }
```

**What did you do? Please include the actual source code causing the issue.**

```js
/* eslint prefer-destructuring: error */

let foo = 0;
foo += bar.foo;
```

**What did you expect to happen?**

No error, I guess.

**What actually happened? Please include the actual, raw output from ESLint.**

<pre>  4:1  error  Use object destructuring                  prefer-destructuring</pre>

**Comment by not-an-aardvark** [comment]
I can reproduce this, thanks for the report.

It's happening because the rule is treating the line the same as

```js
let foo = bar.foo;
```

...which could be fixed to

```js
let {foo} = bar;
```

**Comment by thealjey** [comment]
This rule should only test variable definitions, not all types of assignments.
For example, this line is also a false positive:
```javascript
global.window = jsdom().defaultView;
```

**Comment by not-an-aardvark** [comment]
@thealjey Actually, that looks like it's working as intended -- you can replace that line with

```js
({ defaultView: global.window } = jsdom());
```

**Comment by thealjey** [comment]
@not-an-aardvark Adding parentheses, seriously?
This rule has a big potential to cleanup the code in variable definitions and function arguments.
But instead you're advising to just overcome its shortcomings, to write longer and harder to read code, simply to make a linting rule pass.
Do you really like what you've written, do you think it's an improvement?

**Comment by not-an-aardvark** [comment]
There certainly might be room for improvement in the rule's design -- I was just clarifying that the rule is currently working as intended for your example.

**Comment by thealjey** [comment]
this rule is a very good idea, big 👍  to its author
I would never have come up with something like this myself
it's just that as an average ESLint user I got used to everything always being of the charts awesome
you guys are spoiling us 😃

**Comment by kaicataldo** [comment]
@thealjey Feel free to write up a proposal and submit a new issue if you see something you think could be improved.

**Comment by thealjey** [comment]
@kaicataldo 
I personally am not that big on helping people :) Creating and properly writing an issue just seems like too much work, sorry 😊 

And my suggestion is already out there, it's pretty simple.
It goes like this:

This rule should completely ignore everything but variable definitions (var, const, let, function arguments) and ignore things that are already a destructuring assignment.

That's it, at least I cannot think of any other valid use for this rule.

**Comment by kaicataldo** [comment]
@thealjey Thanks for your input.
